consensus/service/src/mint_tx_manager/mod.rs

```diff
@@ -108,32 +109,25 @@ impl<L: Ledger> MintTxManager for MintTxManagerImpl<L> {
 
     fn combine_mint_config_txs(
         &self,
-        txs: &[MintConfigTx],
+        txs: &[(MintConfigTx, u64)],
         max_elements: usize,
-    ) -> MintTxManagerResult<Vec<MintConfigTx>> {
-        let mut candidates = txs.to_vec();
-        candidates.sort();
-
-        let mut seen_nonces = HashSet::default();
-        let (allowed_txs, _rejected_txs) = candidates.into_iter().partition(|tx| {
-            let nonce_with_token_id = NonceByTokenId {
-                nonce: tx.prefix.nonce.clone(),
-                token_id: tx.prefix.token_id,
-            };
-            if seen_nonces.len() >= max_elements {
-                return false;
-            }
-            if seen_nonces.contains(&nonce_with_token_id) {
-                return false;
-            }
-            seen_nonces.insert(nonce_with_token_id);
-            true
-        });
-
-        Ok(allowed_txs)
+    ) -> MintTxManagerResult<Vec<(MintConfigTx, u64)>> {
+        let mut candidates = timestamp_validator::sort_and_dedup(txs.iter());
+        candidates.truncate(max_elements);
+        Ok(candidates)
```

Comment:
So the idea here is that the nonce needs to be unique per token id - which does mean you can see the same nonce across different token ids. We don't want two instances of the same `(nonce, token_id)` which previously the logic here enforced, but with this change that will no longer be the case since you will be comparing against the entire `MintConfigTx`.
Perhaps `nonce` is a wrong name and it should've been `unique id` or something like that. Does that make sense?

Reply:
Added back logic to prevent having multiple `(nonce, token_id)` in the Mint Config or Txs